packages/react/src/components/Loader/Loader.tsx

```diff
@@ -51,27 +61,61 @@ class Loader extends UIComponent<ReactProps<LoaderProps>> {
       content: false,
       color: true,
     }),
+    delay: PropTypes.number,
     indicator: customPropTypes.itemShorthand,
+    inline: PropTypes.bool,
     label: customPropTypes.itemShorthand,
     labelPosition: PropTypes.oneOf(['above', 'below', 'start', 'end']),
     size: customPropTypes.size,
   }
 
   static defaultProps = {
     accessibility: loaderBehavior,
+    delay: 0,
     indicator: '',
     labelPosition: 'below',
     size: 'medium',
   }
 
+  delayTimer: number
+
+  constructor(props, context) {
+    super(props, context)
+
+    this.state = {
+      visible: this.props.delay === 0,
+    }
+  }
+
+  componentDidMount() {
+    const { delay } = this.props
+
+    if (delay > 0) {
+      this.delayTimer = window.setTimeout(() => {
+        this.setState({ visible: true })
+      }, delay)
+    }
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.delayTimer)
```

Comment:
Check if this.delayTimer is not null/undefined before calling clearTimeout

Reply:
Can you please clarify why need it?

> Passing an invalid ID to clearTimeout() silently does nothing; no exception is thrown.

https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/clearTimeout#Notes

### IE 11

![image](https://user-images.githubusercontent.com/14183168/53476876-1ab5f980-3a7c-11e9-841d-436d21ced9ce.png)

### Chrome

![image](https://user-images.githubusercontent.com/14183168/53476886-22759e00-3a7c-11e9-8b06-9675adb1143b.png)